bundles/org.openhab.ui.habot/src/main/java/org/openhab/ui/habot/notification/internal/webpush/Subscription.java

```diff
@@ -24,21 +24,21 @@
  * @author Martijn Dwars - Initial contribution
  * @author Yannick Schaus - integration for HABot
  */
+@NonNullByDefault
 public class Subscription implements Identifiable<Keys> {
     public String endpoint;
 
-    @NonNull
     public Keys keys;
 
-    public Subscription(String endpoint, @NonNull Keys keys) {
+    public Subscription(String endpoint, Keys keys) {
         this.endpoint = endpoint;
         this.keys = keys;
     }
 
     public class Keys {
-        public String p256dh;
+        public @NonNullByDefault({}) String p256dh;
 
-        public String auth;
+        public @NonNullByDefault({}) String auth;
```

Comment:
Do you know if the default constructor is unnecessary @ghys?
If so, we can remove it and make these fields final without this annotation.

Reply:
Hi @wborn, yes it appears to be unnecessary. It actually comes from https://github.com/web-push-libs/webpush-java/blob/master/src/main/java/nl/martijndwars/webpush/Subscription.java, I don't remember the exact reason why I didn't simply use the library, I think I needed some changes.

Reply:
Gone it is! :+1: